implementations/micrometer-registry-azure/src/main/java/io/micrometer/azure/AzureMeterRegistry.java

```diff
@@ -0,0 +1,301 @@
+     * @param meterName name of the DistributionSummary meter
+     * @param properties dimensions of the metric
+     * @param meter Meter holding rate aggregated metric values
+     */
+    private void trackDistributionSummary(String meterName, Map<String, String> properties,
+        DistributionSummary meter) {
+        MetricTelemetry metricTelemetry = createAndGetBareBoneMetricTelemetry(meterName, properties);
+        metricTelemetry.setValue(meter.totalAmount());
+        metricTelemetry.setCount((int)meter.count());
+        metricTelemetry.setMax(meter.max());
+
+        // TODO: Is setting min to 0th percentile value apt workaround?
+        HistogramSnapshot snapshot = meter.takeSnapshot();
+        Optional<ValueAtPercentile[]> opt = Optional.ofNullable(snapshot.percentileValues());
+        opt.ifPresent(u -> { if (u.length > 0) metricTelemetry.setMin(u[0].value(TimeUnit.MILLISECONDS)); });
+        client.trackMetric(metricTelemetry);
+        logger.trace(String.format("sent metric with name %s", metricTelemetry.getName()));
+
+    }
+
+    /**
+     * Converts Timer type meter to Azure Format and transmits to Azure Monitor backend
+     * @param meterName Name of the Timer Meter
+     * @param properties Dimensions of the metric
+     * @param meter Timer type Meter holding rate aggregated metric values
+     */
+    private void trackTimer(String meterName, Map<String, String> properties, Timer meter) {
+        MetricTelemetry metricTelemetry = createAndGetBareBoneMetricTelemetry(meterName, properties);
+        metricTelemetry.setValue(meter.totalTime(getBaseTimeUnit()));
+        metricTelemetry.setCount((int)meter.count());
+
+        // TODO: Is setting min to 0th percentile value apt workaround?
+        HistogramSnapshot snapshot = meter.takeSnapshot();
+        Optional<ValueAtPercentile[]> opt = Optional.ofNullable(snapshot.percentileValues());
+        opt.ifPresent(u -> { if (u.length > 0) metricTelemetry.setMin(u[0].value(TimeUnit.MILLISECONDS)); });
+        metricTelemetry.setMax(meter.max(getBaseTimeUnit()));
+        client.trackMetric(metricTelemetry);
+        logger.trace(String.format("sent metric with name %s", metricTelemetry.getName()));
+    }
+
+    /**
+     * Converts FunctionTimer tye meter to Azure Format and transmits to Azure Monitor backend
+     * @param meterName Name of the Function Timer
+     * @param properties Dimensions of the metric
+     * @param meter FunctionTimer type meter holding rate aggregated metric values
+     */
+    private void trackTimer(String meterName, Map<String, String> properties, FunctionTimer meter) {
+        MetricTelemetry metricTelemetry = createAndGetBareBoneMetricTelemetry(meterName, properties);
+        metricTelemetry.setValue(meter.totalTime(getBaseTimeUnit()));
+        metricTelemetry.setCount((int)meter.count());
+        client.trackMetric(metricTelemetry);
+        logger.trace(String.format("sent metric with name %s", metricTelemetry.getName()));
```

Comment:
it may help troubleshoot if you will include the type of the metric in the message `sent timer metric with the name`

Reply:
Updated.